File: 169MajorityElement/majorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let map = {}
    let majorityMark = Math.ceil(nums.length / 2)
    for (let num of nums) {
        map[num] = (map[num] || 0) + 1

        if (map[num] == majorityMark) {
            return num
        }
    }
};

console.log(majorityElement([2, 2, 1, 1, 1, 2, 2]))

File: 169MajorityElement/MajorityElement.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int, int> countMap = new Dictionary<int, int>();

        foreach(int num in nums){
            if(countMap.ContainsKey(num)){
                countMap[num]++;
            } else {
                countMap[num] = 1;
            }

            if(countMap[num] > nums.Length / 2){
                return num;
            }
        }
        return Int32.MaxValue;
    }
}